java/datastore/src/main/java/com/google/datastore/v1/client/RemoteRpc.java

```diff
@@ -110,6 +115,28 @@ public InputStream call(String methodName, MessageLite request) throws Datastore
     }
   }
 
+  @VisibleForTesting
+  void setHeaders(MessageLite request, HttpRequest httpRequest) {
+    httpRequest.getHeaders().put(API_FORMAT_VERSION_HEADER, API_FORMAT_VERSION);
+    if (enableE2EChecksum && request != null) {
+      String checksum = EndToEndChecksumHandler.computeChecksum(request.toByteArray());
+      if (checksum != null) {
+        httpRequest.getHeaders().put(EndToEndChecksumHandler.HTTP_REQUEST_CHECKSUM_HEADER,
+            checksum);
+      }
+    }
+  }
+
+  @VisibleForTesting
+  HttpRequestFactory getClient() {
+    return client;
+  }
+
+  @VisibleForTesting
+  static void setSystemEnvE2EChecksum(boolean v) {
```

Comment:
nit: please rename `v`, likely to `enableE2EChecksum`

Reply:
Done